librocksdb_sys/crocksdb/c.cc

```diff
@@ -2979,85 +2980,95 @@ void crocksdb_user_collected_properties_iter_next(
+      return rep.filter_size;
+    case kRawKeySize:
+      return rep.raw_key_size;
+    case kRawValueSize:
+      return rep.raw_value_size;
+    case kNumDataBlocks:
+      return rep.num_data_blocks;
+    case kNumEntries:
+      return rep.num_entries;
+    case kFormatVersion:
+      return rep.format_version;
+    case kFixedKeyLen:
+      return rep.data_size;
+    case kColumnFamilyID:
+      return rep.column_family_id;
   }
   return 0;
 }
 
-const char* crocksdb_table_properties_get_str(crocksdb_table_properties_t* props,
-                                              crocksdb_table_property_t prop, size_t* slen) {
-  auto rep = props->rep_;
+const char* crocksdb_table_properties_get_str(
+    const crocksdb_table_properties_t* props, crocksdb_table_property_t prop,
+    size_t* slen) {
+  const TableProperties& rep = props->rep;
   switch (prop) {
   case kColumnFamilyName:
-    *slen = rep->column_family_name.size();
-    return rep->column_family_name.data();
+    *slen = rep.column_family_name.size();
+    return rep.column_family_name.data();
   case kFilterPolicyName:
-    *slen = rep->filter_policy_name.size();
-    return rep->filter_policy_name.data();
+    *slen = rep.filter_policy_name.size();
+    return rep.filter_policy_name.data();
   case kComparatorName:
-    *slen = rep->comparator_name.size();
-    return rep->comparator_name.data();
+    *slen = rep.comparator_name.size();
+    return rep.comparator_name.data();
   case kMergeOperatorName:
-    *slen = rep->merge_operator_name.size();
-    return rep->merge_operator_name.data();
+    *slen = rep.merge_operator_name.size();
+    return rep.merge_operator_name.data();
   case kPrefixExtractorName:
-    *slen = rep->prefix_extractor_name.size();
-    return rep->prefix_extractor_name.data();
+    *slen = rep.prefix_extractor_name.size();
+    return rep.prefix_extractor_name.data();
   case kPropertyCollectorsNames:
-    *slen = rep->property_collectors_names.size();
-    return rep->property_collectors_names.data();
+    *slen = rep.property_collectors_names.size();
+    return rep.property_collectors_names.data();
   case kCompressionName:
-    *slen = rep->compression_name.size();
-    return rep->compression_name.data();
+    *slen = rep.compression_name.size();
+    return rep.compression_name.data();
   }
   return nullptr;
 }
 
-crocksdb_user_collected_properties_t*
-crocksdb_table_properties_get_user_properties(crocksdb_table_properties_t* props) {
-  return &props->users_;
+const crocksdb_user_collected_properties_t*
+crocksdb_table_properties_get_user_properties(
+    const crocksdb_table_properties_t* props) {
+  return reinterpret_cast<const crocksdb_user_collected_properties_t*>(
+      &props->rep.user_collected_properties);
+}
+
+const char* crocksdb_user_collected_properties_get(
+    const crocksdb_user_collected_properties_t* props, const char* key,
+    size_t klen, size_t* vlen) {
+  auto val = props->rep.find(std::string(key, klen));
```

Comment:
I'm not sure. I prefer stick to C++ 11 for now since rocksdb itself still just require C++ 11. It may be surprising if the wrapper requires a newer version of compiler.